zephyr/src/indigo_api_callback_dut.c

```diff
@@ -1263,8 +1268,14 @@ static int get_mac_addr_handler(struct packet_wrapper *req, struct packet_wrappe
         }
     }
 
+    wpa_s = z_wpas_get_handle_by_ifname(get_wireless_interface());
+    if (!wpa_s) {
+        indigo_logger(LOG_LEVEL_ERROR, "%s: Unable to get wpa_s handle for %s\n", __func__, WIRELESS_INTERFACE_DEFAULT);
+        goto done;
+    }
+
     if (atoi(role) == DUT_TYPE_STAUT) {
-        w = wpa_ctrl_open(get_wpas_ctrl_path());
+        w = wpa_ctrl_open(wpa_s->ctrl_iface->sock_pair[0]);
```

Comment:
if you are removing close, why do we need to open everytime? Just use the global variable or add a wrapper to get the handle

Reply:
@krish2718 As per the discussion, keeping the changes which are working fine.